packages/zudoku/src/lib/authentication/authentication.ts

```diff
@@ -6,6 +6,34 @@ export interface AuthenticationPlugin {
   getAccessToken(): Promise<string>;
   onPageLoad?(): void;
   signRequest(request: Request): Promise<Request>;
+  /**
+   * Optional method to render custom sign-in UI.
+   * If provided along with hasCustomUI=true, this will be rendered instead of redirecting.
+   * Returns a component function that renders the custom UI.
+   */
+  renderSignInUI?(): () => React.JSX.Element | null;
+  /**
+   * Optional method to render custom sign-up UI.
+   * If provided along with hasCustomUI=true, this will be rendered instead of redirecting.
+   * Returns a component function that renders the custom UI.
+   */
+  renderSignUpUI?(): () => React.JSX.Element | null;
+  /**
+   * Indicates whether this provider uses custom UI instead of redirects.
+   * If true, the SignIn/SignUp components will render the custom UI.
+   */
+  hasCustomUI?: boolean;
```

Comment:
```suggestion
```

I suggest we remove this flag and just by having the `render` methods, Zudoku will do the custom UI. 
Otherwise you may set this flag to `true` without providing the methods.